sys/action.js

```diff
@@ -137,7 +137,14 @@ function buildQueryResponse(res) {
     } else if (!res.body || res.body.error) {
         throw apiError((res.body || {}).error);
     } else if (!res.body.query || (!res.body.query.pages && !res.body.query.userinfo)) {
-        throw apiError({ info: 'Missing query pages from the PHP action API response.' });
+        throw new HTTPError({
+            status: 404,
+            body: {
+                type: 'not_found#page_revisions',
```

Comment:
Yeah, the `#page_revisions` part is not entirely accurate. Perhaps we could make it something like `#action_query_pages`?